### PR TITLE
Command-a functionality for select/deselect all points

### LINKED ISSUE #248: Command-A to select all points and components
Involves building a new selection object. Should also update the window location.

### PR DESCRIPTION
**Pull request by nedkamburov**
Fixes #248

### CODE REVIEW

File: src/fontra/views/editor/editor.js [hunk 1]
```diff
@@ -714,6 +722,26 @@ export class EditorController {
     );
   }
 
+  async doSelectAllNone(selectNone) {
+    const positionedGlyph =
+      await this.sceneController.sceneModel.getSelectedPositionedGlyph();
+
+    if (positionedGlyph) {
+      const glyphPath = positionedGlyph.glyph.path;
+      const allPoints = new Set();
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:731]: Call this variable `newSelection`

File: src/fontra/views/editor/editor.js [hunk 2]
```diff
@@ -714,6 +722,26 @@ export class EditorController {
...
+
+      for (const [pointIndex] of glyphPath.pointTypes.entries()) {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:733]: Do this only if `!selectNone`

File: src/fontra/views/editor/editor.js [hunk 3]
```diff
@@ -714,6 +722,26 @@ export class EditorController {
...
+        allPoints.add(`point/${pointIndex}`);
+      }
+
+      this.sceneController.selection = allPoints;
+    }
+
+    if (selectNone) {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:740]: ...and get rid of this block, should be handled above.

File: src/fontra/views/editor/editor.js (hunk 2 above)
Comment (justvanrossum) [reply src/fontra/views/editor/editor.js:733]: (Good use of `.entries()`! I didn't think of that.)

Comment (justvanrossum) [reply src/fontra/views/editor/editor.js:733]: In the same `!selectNone` block, also add a loop that adds all `component/${componentIndex}` indices.

**Comment by nedkamburov** [comment]
I made the changes. I hope adding glyphPath points and glyphComponents covers all cases?

**Review by justvanrossum — APPROVED** [review]
Perfect, thanks!

**Pull request merged**